fletcher/base.py

```diff
@@ -637,6 +641,9 @@ def __getitem__(self, item):
         For a boolean mask, return an instance of ``ExtensionArray``, filtered
         to the values where ``item`` is True.
         """
+        if PANDAS_GE_0_26_0 and is_bool_dtype(item):
+            check_bool_array_indexer(self, item)
```

Comment:
What it this doing?

Reply:
See https://github.com/pandas-dev/pandas/pull/30308. It is checking nullable boolean arrays for NAs (and raise if there are), and return a numpy array. 

So, actually, this needs to re-assign to `item`.